http/http-advanced/src/main/java/io/quarkus/ts/http/advanced/GrpcResource.java

```diff
@@ -1,18 +1,24 @@
 package io.quarkus.ts.http.advanced;
 
+import java.util.HashSet;
+import java.util.Set;
+
 import jakarta.inject.Inject;
 import jakarta.ws.rs.GET;
 import jakarta.ws.rs.Path;
 import jakarta.ws.rs.PathParam;
 import jakarta.ws.rs.Produces;
 import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
 
 import io.quarkus.example.GreeterGrpc;
+import io.quarkus.example.HelloReply;
 import io.quarkus.example.HelloRequest;
 import io.quarkus.grpc.GrpcClient;
 
 @Path("/grpc")
 public class GrpcResource {
+    static Set<String> invoked = new HashSet<>();
```

Comment:
My impression is that this approach is spreading from previous PRs, but these kind of data should be passed along with request / response. Only reason why you can do it is because you call same gRPC server. But in production you will never use gRPC client to call anything inside very same application.

I know this can be dogmatic, but I'd like to follow approach closer to what users will do, especially because this way indirectly we test other things, not just isolated feature.

I'll leave below suggestions how to do this.